### DISCUSSION TITLE
Feature request: Add support for multi-column view of events

### DISCUSSION THREAD

**Discussion by DanAvni** [datalust/seq-tickets#1641] [Ideas]
Trying to debug a multi-threaded/multi-process event list in one browser instance is very confusing as different events belong to different processes/threads and they are all mixed (I do need to see them in one browser to understand the event sequence). I find myself constantly expanding/collapsing the events to see the process they belong to.

My suggestion: Imagine the text of the event as a column and I could add more columns and specify for each one a filter. An event will be displayed on the list in a column with a filter for it or in the 1st column if no match is found (1st column has no filter). That way I could get to display something like the mock screenshot I created (attached) which I believe could allow for a better understanding of these situations.

Another great related feature could be to auto divide into columns based on specific criteria I select. So I could say I want a column for each unique process & thread so I would get as many columns as needed to display the various processes and within each one, the different threads. A manual option I think is also required if I want to specify complex criteria for each column like I described earlier.
![HsB9N17 1](https://user-images.githubusercontent.com/4114116/186080629-76bf24f7-1b6d-45fe-bea4-0b9b9dc8b18e.png)

**Comment by liammclennan** [comment]
Hi Dan,

Thanks for the suggestion. We have discussed doing something like this, to show the temporal relationship between interleaved processes. For now, one option is to use the 'Show as column' command on your `ProcessId` property, as I have done here with `RequestPath`. That way you can clearly see which events belong to which process. 

![image](https://user-images.githubusercontent.com/271514/186084024-a7b52858-450d-4819-8373-5054aeb975c5.png)

**Reply by DanAvni** [reply]
I was not aware of this feature and indeed it helps thanks! I still think though that a columnar view would be better as it allows me to put into columns things I want the way I logically think of them (each column with it's own filter) as opposed to viewing all info columns in a flat list and having to split the events in my head
